src/user_saved/views.py

```diff
@@ -0,0 +1,224 @@
+from django.db import IntegrityError
+from django.db.models import Count
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from researchhub_document.related_models.researchhub_unified_document_model import (
+    ResearchhubUnifiedDocument,
+)
+from researchhub_document.serializers.researchhub_unified_document_serializer import (
+    DynamicUnifiedDocumentSerializer,
+)
+from researchhub_document.views.researchhub_unified_document_views import (
+    ResearchhubUnifiedDocumentViewSet,
+)
+
+from .models import UserSavedEntry, UserSavedList
+from .serializers import (
+    ChangeDocumentSerializer,
+    CreateListSerializer,
+    DeleteListSerializer,
+    UserSavedListSerializer,
+)
+
+LIST_NOT_FOUND = "List not found"
+
+
+class UserSavedView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+        input_list_name = request.query_params.get("list_name")
+        u_doc_id = request.query_params.get("u_doc_id")
+        paper_id = request.query_params.get("paper_id")
+        all_flag = request.query_params.get("all_items")
+        if all_flag:
+            docs = ResearchhubUnifiedDocument.objects.filter(
+                usersavedentry__created_by=request.user,
+                usersavedentry__is_removed=False,
+            ).annotate(count=Count("usersavedentry"))
+            res = {str(doc.id): doc.count for doc in docs}
+            return Response(res, status=status.HTTP_200_OK)
```

Comment:
Shouldn't this be a separate endpoint? The shape of the data is totally different, so adding the querystring is not filtering, but changing the response type.